### ISSUE TITLE
Bug Report:  Jakarta EE 9 Form Data

### ISSUE THREAD

**Issue by hink084** [payara/Payara#5842]
# Description #
----------

I started upgrading from Jakarta EE 8 to EE 9.1 using Payara 5.2022.2 and noticed the properites submitted with the form data for an XMLHttpRequest are incorrect or missing. Payara 5.2022.2 should be Jakarta EE 9 compatible according to the release documentation.  I have attached a sample application to demo the issues - javax is utilized instead of jakarta or jakarta.faces.ViewState is missing.  Without making any configuration changes, I deployed the same application to Glassfish 6.2.5 and the properties submitted with the form data are correct.  I was unable to test with Payara 6.2022.1 Alpha 2.

The sample application contains an h:inputText, h:commandButton, p:commandButton, p:commandButton, and an h:outputText.  The expected and acutal outcomes when each button is clicked are defined below. Sample input prior to each button click is respectively: Test 1, Test 2, Test 3

**index.xhtml**
```
<h:form id="form">
    <h:inputText id="input"
                 value="#{index.value}"/>
    &nbsp;
    <h:commandButton id="jsfButton"
                     value="JSF">
        <f:ajax execute="input"
                render="output"/>
    </h:commandButton>
    &nbsp;
    <p:commandButton id="partialPrimeButton"
                     partialSubmit="true"
                     process="input"
                     update="output"
                     value="Partial PrimeFaces"/>
    &nbsp;
    <p:commandButton id="primeButton"
                     process="input"
                     update="output"
                     value="PrimeFaces"/>
    <br/>
    Message: <h:outputText id="output"
                           value="#{not empty index.value ? index.value : '- Empty -'}"/>
</h:form> 
``` 

**index.java**
```
@Named
@ViewScoped
public class Index implements Serializable{
    private String value;

    public String getValue(){
        return value;
    }

    public void setValue(String value){
        this.value = value;
    }
}
```

**web.xml**
```
<web-app xmlns="https://jakarta.ee/xml/ns/jakartaee"
         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="https://jakarta.ee/xml/ns/jakartaee https://jakarta.ee/xml/ns/jakartaee/web-app_5_0.xsd"
         version="5.0">
    <context-param>
        <param-name>jakarta.faces.PROJECT_STAGE</param-name>
        <param-value>Development</param-value>
    </context-param>
    <context-param>
        <param-name>jakarta.faces.STATE_SAVING_METHOD</param-name>
        <param-value>server</param-value>
    </context-param>

    <servlet>
        <servlet-name>Faces Servlet</servlet-name>
        <servlet-class>jakarta.faces.webapp.FacesServlet</servlet-class>
        <load-on-startup>1</load-on-startup>
    </servlet>

    <servlet-mapping>
        <servlet-name>Faces Servlet</servlet-name>
        <url-pattern>*.xhtml</url-pattern>
    </servlet-mapping>

    <session-config>
        <session-timeout>
            30
        </session-timeout>
    </session-config>

    <welcome-file-list>
        <welcome-file>index.xhtml</welcome-file>
    </welcome-file-list>
</web-app>
```

## Expected Outcome ##

**JSF Button Click**
```
form: form
form:input: Test
jakarta.faces.behavior.event: action
jakarta.faces.partial.ajax: true
jakarta.faces.partial.event: click
jakarta.faces.partial.execute: form:jsfButton form:input
jakarta.faces.partial.render: form:output
jakarta.faces.source: form:jsfButton
jakarta.faces.ViewState: -6775913271899001914:-5499159218107670136
```

**Partial PrimeFaces Button Click**
```
form:input: Test 2
form:partialPrimeButton: form:partialPrimeButton
jakarta.faces.partial.ajax: true
jakarta.faces.partial.execute: form:input
jakarta.faces.partial.render: form:output
jakarta.faces.source: form:partialPrimeButton
jakarta.faces.ViewState: -6775913271899001914:-5499159218107670136
```

**PrimeFaces Button Click**
```
form: form
form:input: Test 3
form:primeButton: form:primeButton
jakarta.faces.partial.ajax: true
jakarta.faces.partial.execute: form:input
jakarta.faces.partial.render: form:output
jakarta.faces.source: form:primeButton
jakarta.faces.ViewState: -6775913271899001914:-5499159218107670136
```

## Current Outcome ##

**JSF Button Click**
- javax is utilized instead of jakarta
```
form: form
form:input: Test
javax.faces.behavior.event: action
javax.faces.partial.ajax: true
javax.faces.partial.event: click
javax.faces.partial.execute: form:jsfButton form:input
javax.faces.partial.render: form:output
javax.faces.source: form:jsfButton
javax.faces.ViewState: 3768650063610317665:-4574714117490412382
```

**Partial PrimeFaces Button Click**
- Message text is not updated
- jakarta.faces.ViewState is missing
```
form:input: Test 2
form:partialPrimeButton: form:partialPrimeButton
jakarta.faces.partial.ajax: true
jakarta.faces.source: form:partialPrimeButton
jakarta.faces.partial.execute: form:input
jakarta.faces.partial.render: form:output
```

**PrimeFaces Button Click**
- Message text is not updated
- javax.faces.ViewState is utilized instead of jakarta.faces.ViewState
```
form: form
form:input: Test 3
form:primeButton: form:primeButton
jakarta.faces.partial.ajax: true
jakarta.faces.partial.execute: form:input
jakarta.faces.partial.render: form:output
jakarta.faces.source: form:primeButton
javax.faces.ViewState: 3768650063610317665:-4574714117490412382
```

## Steps to reproduce ## 

1) Clean, Build, and Run the attached project.  (Maven NetBeans project)
2) Press F12 to open the Browser Developer Tools - Network Headers/Payload
3) Type in the inputText and click the submit buttons
4) Observe payload values

## Environment ##

- **Distribution**: Payara Community 5.2022.2 Build 306
- **JDK Version**: Eclipse Adoptium 11.0.13+8
- **Operating System**: Windows 10
- **Build Tool**: Maven 3.8.5

## Attached Sample Project ##
[Jakarta9.zip](https://github.com/payara/Payara/files/9087693/Jakarta9.zip)

**Comment by poikilotherm** [comment]
I might have run into a similar issue while hacking on https://github.com/ocpsoft/rewrite/pull/350

When using a `h:outputScript`, the generated resource path contains `javax.faces.resources` instead of `jakarta.faces.resources`. Do you want me to create a simple reproducer? (It should be easy to create a stripped down version of the test and it's classes.)

**Comment by shub8968** [comment]
Hi, @hink084 & @poikilotherm.

Jakarta EE 9.x applications are supported on Payara 5 by the integrated Eclipse Transformer.  At deployment time, the code binaries are transformed (jakarta → javax) and descriptors are updated.

**Comment by hink084** [comment]
Thanks for the info @shub8968, I did not realize the direction of the Eclipse Transformer on Payara 5 (jakarta &#8594; javax).  

I recently discovered a couple of Payara blog articles from 2021 that mention this topic for anyone that is interested:

- [Payara Server Community 6.2021.1.Alpha1 is now a Certified Jakarta EE 9.1 (Full Profile) Compatible Product](https://blog.payara.fish/payara-server-community-6.2021.1.alpha1-certified-jakarta-ee-9.1-compatible)

- [How to Use Eclipse Transformer to Convert a 3rd Party Library to the New Jakarta Namespace](https://blog.payara.fish/how-to-use-the-eclipse-transformer-project-to-convert-a-3rd-party-library)

I also did not realize Payara 5.2021.3+ was not fully certified for Jakarta EE 9.x.  I saw Jakarta EE 9 listed in the _Supported APIs and Applications_ section of the _Release Notes_ which led me to believe it was certified. I now see the separate _Jakarta EE Certification_ section - plain as day, right below the _Release Notes_ :).

- [Payara Community Documentation](https://docs.payara.fish/community/docs)

I was able to test my submitted reproducer using _Payara 6.2021.1 Alpha 4 Build 133_ and everything works as expected.  I will wait for a production release of Payara 6 to upgrade Jakarta.
